pyfr/backends/cuda/generator.py

```diff
@@ -12,9 +12,8 @@ def __init__(self, *args, **kwargs):
             self._ix = 'int _x = blockIdx.x*blockDim.x + threadIdx.x;'
             self._limits = 'if (_x < _nx)'
         else:
-            self._ix = ('int _x = blockIdx.x*blockDim.x + threadIdx.x;'
-                        'int _y = blockIdx.y*blockDim.y + threadIdx.y;')
-            self._limits = 'if (_x < _nx && _y < _ny)'
+            self._ix = 'int _x = blockIdx.x*blockDim.x + threadIdx.x;'
```

Comment:
_ix can be placed outside of the if statement. Same applies for opencl and hip backends

Reply:
Fixed.